### PR TITLE
Start moving old test suite over to Karma runner

### PR DESCRIPTION
**Pull request by hatched**
The old test suite was showing its age. This branch creates a new karma test runner separate from the 'squeaky clean' one and moves the majority of the old tests over. The ones which weren't able to be moved over because of unsolved failures have been left to run with the old suite.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

If CI passes, and it's actually run the new runner then we're good!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4073/
Test PASSed.

File: Makefile [hunk 1]
```diff
@@ -351,12 +351,17 @@ test: $(JUJUGUI) $(PYTEST)
...
+	./scripts/test-js-old.sh
 
 .PHONY: test-js-karma
 test-js-karma: gui
 	$(NODE_MODULES)/.bin/karma start karma.conf.js --single-run --browsers PhantomJS --log-level warn --reporters mocha
 
+.PHONY: test-old-js-karma
+test-old-js-karma: gui
+	./scripts/test-js.sh
+
```
Comment (bac) [inline Makefile]: OK, this confuses me.  The test-ja-phantom target runs the test-js-old script but the test-old-js-karma target runs the test-js script.  Wha?

Comment (bac) [inline Makefile]: Also 'old' is a terrible name unless it will be going away very quickly, like one week.

**Comment by bac** [comment]
CI passes so that's great.  I've made some comments about the confusing naming that I hope you'll fix or explain.

**Comment by bac** [comment]
Thanks @hatched that's much more understandable.  :+1:

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4076/
Test FAILed.

**Comment by makyo** [comment]
:+1:!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4077/
Test PASSed.

**Comment by hatched** [comment]
Thanks! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/1400

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/1401

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4080/
Test PASSed.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/1403

**Comment by bac** [comment]
@hatched The Makefile has some issues that come to light while I was looking into the CI failures.  Could you make these changes please.

```
diff --git a/Makefile b/Makefile
index d1491b8..74f9e7f 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ GUISRC := jujugui/static/gui/src
 GUIBUILD := jujugui/static/gui/build
 OLD_TEMPLATES_FILE := $(GUIBUILD)/app/templates.js
 SVG_SPRITE_DIR := $(GUIBUILD)/app/assets
-SVG_SPRITE_FILE := $(SVG_SPRITE_FILE)/stack/svg/sprite.css.svg
+SVG_SPRITE_FILE := $(SVG_SPRITE_DIR)/stack/svg/sprite.css.svg
 SVG_SPRITE_SOURCE_DIR := $(GUISRC)/app/assets/svgs
 STATIC_CSS := $(GUIBUILD)/app/assets/css
 STATIC_IMAGES := $(GUIBUILD)/app/assets/images
@@ -54,7 +54,6 @@ help:
 	@echo "bumpversion - bump version."
 	@echo "  By default bumps patch level. 'VPART=[major|minor|patch] make bumpversion' to specify."
 	@echo "check - run tests and check lint."
-	@echo "clean - remove build and python artifacts"
 	@echo "clean-downloadcache - remove the downloadcache"
 	@echo "clean-gui - clean the built gui js code"
 	@echo "deps - install the dependencies"
@@ -208,6 +207,7 @@ $(HANDLEBARS_ASSETS): $(NODE_MODULES)

 $(BUILT_YUI): $(YUI) $(BUILT_JS_ASSETS)
 	cp -r $(YUI) $(BUILT_YUI)
+	touch $(BUILT_YUI)

 $(BUILT_D3):
 	mkdir -p $(GUIBUILD)/app/assets/javascripts
```

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4081/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4082/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4083/
Test PASSed.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by bac** [comment]
@hatched, after your merge job ran the state of the workspace is:
```
jenkins@juju-machine-0-lxc-2:~/workspace/juju-gui-merge$ git status
HEAD detached from 3bd35a9
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   karma-mocha-phantom.conf.js

no changes added to commit (use "git add" and/or "git commit -a")
```

I'm curious to see what happens the next time a job runs.  Won't the git commands that try to setup the new branch fail?

**Pull request merged**